### ISSUE TITLE
Data structure to query for x y positions

### ISSUE THREAD

**Issue by nmcobb** [NGCP/VTOL#180]
Requirements:
O(log(N)) access time
O(log(N)) storage time
Negative indices are accepted
Can efficiently query[xmin:xmax, ymin:ymax]

**Comment by jyee27** [comment]
Some notes from my research:
- An R tree would perform well for range queries. GeoPandas and libspatialindex have R tree implementations. GeoPandas is on conda and pip, while libspatialindex is only on conda.
- SciPy has a kd-tree implementation, but we would need all the points before we could make the tree.

**Comment by nmcobb** [comment]
Yes! These look like exactly what we're looking for. I why don't we go with GeoPandas, that seems like it would be well tailored to our problem. If you have time, you should start playing around with that a bit, maybe just reading on dummy locations for now. Nice job

**Comment by jyee27** [comment]
I made a version using GeoPandas, but ran into trouble with the dependencies. GeoPandas requires Rtree, which requires libspatialindex. I think we should go with the version without GeoPandas from my current pull request.
